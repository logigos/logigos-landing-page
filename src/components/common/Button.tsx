import React, { FC, ReactNode } from 'react'
import styled from 'styled-components'

const ButtonRoot = styled.button<Props>`
    appearance: none;
    /* browser not support yet */

    cursor: pointer;
    display: flex;
    justify-content: center;
    font-family: 'Mitr';
    line-height: 1.8;
    padding: 0;

    border-width: ${props => props.border_width || '0'};
    border-color: ${props => props.border_color || 'white'};
    border-radius: ${props => props.border_radius || '0.8rem'};
    border: ${props => props.border || null};
    margin: ${props => props.margin || '1rem 2.2rem'};
    font-size: ${props => props.font_size || '3.6rem'};
    background: ${props => props.bg || 'white'};
    color: ${props => props.color || 'white'};
    width: ${props => props.width || '27.8rem'};
    height: ${props => props.height || '6.6rem'};

    &:hover {
        background: ${props => props.hbg || 'white'};
        color: ${props => props.hcolor || props.color || 'white'};
        border: ${props => props.hborder || null};
    }

    &:active {
        background: ${props => props.abg || 'white'};
        color: ${props =>
            props.acolor || props.hcolor || props.color || 'white'};
        border: ${props => props.aborder || null};
    }

    & a {
        text-decoration: none;
        color: currentColor;
    }
`

interface Props {
    children?: ReactNode
    width?: string
    height?: string
    font_size?: string
    margin?: string
    color?: string
    hcolor?: string
    acolor?: string
    bg?: string
    hbg?: string
    abg?: string
    border_radius?: string
    border_width?: string
    border_color?: string
    border?: string
    aborder?: string
    hborder?: string
}

const Button: FC<Props> = ({ children, ...props }) => {
    return <ButtonRoot {...props}>{children}</ButtonRoot>
}

export default Button
