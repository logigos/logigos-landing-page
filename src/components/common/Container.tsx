// import React from 'react'
import styled from 'styled-components'

export const Conflexrow = styled.div`
    display: flex;
`

export const ConflexrowCenter = styled.div`
    display: flex;
    justify-content: center;
    align-items: center;
`

export const ConflexcolCenter = styled(ConflexrowCenter)`
    flex-direction: column;
`

export const Headerflexrow = styled.header`
    display: flex;
`

export const Conflexcol = styled(Conflexrow)`
    flex-direction: column;
`

export const ConflexcolCenterH = styled(Conflexcol)`
    align-items: center;
`

export const ContainerSection = styled(Conflexcol)`
    align-items: center;
    margin-top: 8.8rem;
`
