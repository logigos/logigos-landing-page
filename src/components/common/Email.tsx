import React from 'react'
import styled from 'styled-components'
import R from '../resources/R'
import { TextSectionA } from './Text'
import IconEMail from '../../images/contact-section/icon/email.svg'

const WrapEmail = styled.div`
    cursor: pointer;
    display: flex;
    align-items: center;
`

const IconEMailCustom = styled(IconEMail)`
    ${WrapEmail}:hover & {
        path {
            fill: ${R.colors.base_red_l_75};
        }
    }

    ${WrapEmail}:active & {
        path {
            fill: ${R.colors.base_red_l_60};
        }
    }
`

const TextEmail = styled(TextSectionA)`
    font-size: 2.4rem;
    color: ${R.colors.white};
    font-weight: 400;
    margin: auto 0 auto 1.6rem;
    height: 4rem;
    box-sizing: border-box;

    ${WrapEmail}:hover & {
        border-bottom: 0.3rem solid ${R.colors.base_red_l_75};
    }
    ${WrapEmail}:active & {
        border-bottom: 0.3rem solid ${R.colors.base_red_l_60};
    }
`

const Email = () => {
    return (
        <WrapEmail>
            <IconEMailCustom />
            <TextEmail
                href="mailto:logigos.official@gmail.com"
                target="_blank"
                rel="noopener noreferrer"
            >
                logigos.official@gmail.com
            </TextEmail>
        </WrapEmail>
    )
}
export default Email
