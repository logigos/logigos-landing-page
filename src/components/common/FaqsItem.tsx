import React, { FC } from 'react'
import styled, { css } from 'styled-components'
import R from '../resources/R'
import IconArrow from '../../images/faq-section/icon/arrow-down-blue.svg'

const IconArrowCustom = styled(IconArrow)`
    position: absolute;
    top: 2.8rem;
    right: 3.88rem;
`

const QText = styled.p`
    font-family: 'Mitr';
    height: 3.8rem;
    font-size: 2.4rem;
    color: ${R.colors.black_l_40};
    margin: 1.6rem 0 1rem 0;
`

const AText = styled.p`
    width: 55.7rem;
    max-height: 10.9rem;
    font-family: 'Mitr';
    font-size: 2.4rem;
    color: ${R.colors.black_l_60};
    /* line-height: 1.8em; */
    margin: 0 0 1.9rem 0;
`
export interface FAQType {
    Q: string
    A: string
    expand: boolean
    handleOnClick: () => void
}

const FaqsItem: FC<FAQType> = ({ Q, A, expand, handleOnClick }) => {
    return (
        <ContainerItem onClick={() => handleOnClick()} isExpand={expand}>
            <IconArrowCustom />
            <QText>{Q}</QText>
            <AText>{A}</AText>
        </ContainerItem>
    )
}

const ContainerItem = styled.div<{ isExpand: boolean }>`
    position: relative;
    width: 100%;
    /* height: 7rem; */
    max-height: 7rem;
    margin: 0.6rem 0;
    overflow: hidden;
    border-radius: 8px;
    background: ${R.colors.base_blue_l_96};
    padding-left: 3.6rem;
    cursor: pointer;

    ${props =>
        props.isExpand
            ? css`
                  max-height: 19.2rem;
                  transition: max-height 300ms ease-out;
              `
            : css`
                  max-height: 7rem;
                  transition: max-height 300ms ease-out;
              `}
`

export default FaqsItem
