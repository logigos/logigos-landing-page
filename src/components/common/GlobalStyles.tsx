import { createGlobalStyle } from 'styled-components'

export default createGlobalStyle`

html {
    font-size: 62.5%;
  }

  body{
    margin: 0;
    font-family: Kanit-Regular;
    font-size: 1.6rem; /* 16px */
  }
`
