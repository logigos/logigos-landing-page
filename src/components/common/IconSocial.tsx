import styled, { css } from 'styled-components'
import R from '../resources/R'
import IconFaceBookInLine from '../../images/contact-section/icon/facebook.svg'
import IconLineInLine from '../../images/contact-section/icon/line.svg'

const IconSocial = css`
    cursor: pointer;
    margin: auto 0 auto 4.3rem;
    box-sizing: border-box;
    border-radius: 50%;
    border: 5px solid transparent;

    &:hover {
        border-color: ${R.colors.base_red_l_75};
    }
    &:active {
        border-color: ${R.colors.base_red_l_60};
    }
`
interface IconSocialType {
    width?: string
    height?: string
}

export const IconFaceBook = styled(IconFaceBookInLine)<IconSocialType>`
    ${IconSocial}
    width: ${props => props.width || '6.6rem'};
    height: ${props => props.height || '6.6rem'};
`

export const IconLine = styled(IconLineInLine)<IconSocialType>`
    ${IconSocial}
    width: ${props => props.width || '6.6rem'};
    height: ${props => props.height || '6.6rem'};
`
