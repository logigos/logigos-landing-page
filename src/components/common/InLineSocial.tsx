import React, { FC } from 'react'
import styled, { css } from 'styled-components'
import R from '../resources/R'
import { TextSectionA } from './Text'
import IconEMail from '../../images/contact-section/icon/email.svg'
import IconTell from '../../images/footer-section/icon/tel.svg'

const Wrap = styled.div<{ margin?: string }>`
    cursor: pointer;
    display: flex;
    align-items: center;
    margin: ${props => props.margin || '0'};
`

const RulesIconSocial = css`
    ${Wrap}:hover & {
        path {
            fill: ${R.colors.base_red_l_75};
        }
    }

    ${Wrap}:active & {
        path {
            fill: ${R.colors.base_red_l_60};
        }
    }
`

const IconEMailCustom = styled(IconEMail)`
    ${RulesIconSocial}
`

const IconTellCustom = styled(IconTell)`
    ${RulesIconSocial}
`

const TextLine = styled(TextSectionA)`
    cursor: pointer;
    font-family: 'Mitr';
    font-size: 2.4rem;
    color: ${R.colors.white};
    font-weight: 400;
    margin: auto 0 auto 1.6rem;
    height: 4rem;
    box-sizing: border-box;

    ${Wrap}:hover & {
        border-bottom: 0.3rem solid ${R.colors.base_red_l_75};
    }
    ${Wrap}:active & {
        border-bottom: 0.3rem solid ${R.colors.base_red_l_60};
    }
`

interface IconSocialType {
    width?: string
    height?: string
    margin?: string
}

export const LineEmail: FC<IconSocialType> = props => {
    return (
        <Wrap margin={props.margin}>
            <IconEMailCustom width={props.width} height={props.height} />
            <TextLine
                href="mailto:logigos.official@gmail.com"
                target="_blank"
                rel="noopener noreferrer"
            >
                logigos.official@gmail.com
            </TextLine>
        </Wrap>
    )
}

export const LineTell = () => {
    return (
        <Wrap>
            <IconTellCustom />
            <TextLine href="tel:0873370808">00 000 0000</TextLine>
        </Wrap>
    )
}
