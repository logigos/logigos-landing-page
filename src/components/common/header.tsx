import React from 'react'
import styled from 'styled-components'
import { ContainerFlexRow, ContainerFlexCol } from './Container'

const ContainerHeader = styled(ContainerFlexRow)`
    width: 100%;
`

const ContainerContent = styled(ContainerFlexCol)`
    margin-top: 14.2rem;
    margin-left: 15.6rem;
    background: orange;
`

const Logigos = styled.h1`
    font-family: 'Montserrat';
    font-size: 6.4rem;
    font-weight: 800; /* ExtraBlod */
    color: white;
    margin-bottom: 3.2rem;
`

const Detail = styled.p`
    font-family: 'Mitr';
    width: 48.3rem;
    display: block;
    text-overflow: ellipsis;
    word-wrap: break-word;
    overflow: hidden;
    font-size: 3.2rem;
    max-height: 10rem;
    line-height: 5rem;
`

const Header = () => {
    return (
        <ContainerHeader>
            <ContainerContent>
                <Logigos>Logigos</Logigos>
                <Detail>
                    ค้นหารถบรรทุก งานขนส่งสินค้า
                    <br /> และจัดการระบบขนส่งอย่างมีคุณภาพ
                </Detail>
            </ContainerContent>
        </ContainerHeader>
    )
}

export default Header
