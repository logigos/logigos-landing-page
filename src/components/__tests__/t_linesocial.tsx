import React from 'react'
import renderer from 'react-test-renderer'

import { LineEmail, LineTell } from '../common/InLineSocial'

describe('LineSocial', () => {
    it('LineEmail | renders correctly', () => {
        const tree = renderer
            .create(<LineEmail width="4rem" height="4rem" />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('LineTell | renders correctly', () => {
        const tree = renderer.create(<LineTell />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
