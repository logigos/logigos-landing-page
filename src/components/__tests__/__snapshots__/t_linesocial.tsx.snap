// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineSocial LineEmail | renders correctly 1`] = `
<div
  className="sc-AxhCb dhpJBI"
>
  <test-file-stub
    className="sc-AxhUy bTNPmY"
    height="4rem"
    width="4rem"
  />
  <a
    className="sc-AxiKw sc-AxheI iEaKEL"
    href="mailto:logigos.official@gmail.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    logigos.official@gmail.com
  </a>
</div>
`;

exports[`LineSocial LineTell | renders correctly 1`] = `
<div
  className="sc-AxhCb dhpJBI"
>
  <test-file-stub
    className="sc-AxgMl jhutUO"
  />
  <a
    className="sc-AxiKw sc-AxheI iEaKEL"
    href="tel:0873370808"
  >
    00 000 0000
  </a>
</div>
`;
